In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
data=pd.read_csv('bbq_summer.csv')

In [2]:
data

bundleID                starter                         maindishI  \
0           1  Fried Chicken Tenders                       BBQ Brisket   
1           2  Fried Chicken Tenders                       BBQ Brisket   
2           3  Fried Chicken Tenders                       BBQ Brisket   
3           4  Fried Chicken Tenders                       BBQ Brisket   
4           5  Fried Chicken Tenders                       BBQ Brisket   
..        ...                    ...                               ...   
379       380  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
380       381  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
381       382  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
382       383  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   
383       384  Jumbo Shrimp Cocktail  Buffalo Chicken Macaroni Stacker   

        maindishII                side              dessert  avg_rating  
0          Sausage      Mac and Cheese      Blondie Brownie       6.040  
1          Sausage      Mac and Cheese  Apple Pie a la Mode       9.290  
2          Sausage        Potato Salad      Blondie Brownie       6.200  
3          Sausage        Potato Salad  Apple Pie a la Mode       8.840  
4          Sausage  French Fry Platter      Blondie Brownie       8.240  
..             ...                 ...                  ...         ...  
379  Steak Sampler      Mac and Cheese  Apple Pie a la Mode       4.738  
380  Steak Sampler        Potato Salad      Blondie Brownie       4.768  
381  Steak Sampler        Potato Salad  Apple Pie a la Mode       6.248  
382  Steak Sampler  French Fry Platter      Blondie Brownie       7.138  
383  Steak Sampler  French Fry Platter  Apple Pie a la Mode       7.428  

[384 rows x 7 columns]

In [3]:
data.columns

Index(['bundleID', 'starter', 'maindishI', 'maindishII', 'side', 'dessert',
       'avg_rating'],
      dtype='object')

In [4]:
data=data.drop(columns=['bundleID'])

In [5]:
data_d=pd.get_dummies(data, drop_first=True,columns=['starter','maindishI','maindishII','side','dessert'])
data_d.head()

avg_rating  starter_Fried Chicken Tenders  starter_Jumbo Shrimp Cocktail  \
0        6.04                              1                              0   
1        9.29                              1                              0   
2        6.20                              1                              0   
3        8.84                              1                              0   
4        8.24                              1                              0   

   starter_Sticky Chicken Tenders  maindishI_BBQ Chicken  \
0                               0                      0   
1                               0                      0   
2                               0                      0   
3                               0                      0   
4                               0                      0   

   maindishI_Buffalo Chicken Macaroni Stacker  maindishI_Pulled Pork  \
0                                           0                      0   
1                                           0                      0   
2                                           0                      0   
3                                           0                      0   
4                                           0                      0   

   maindishII_Lobster Roll  maindishII_Sausage  maindishII_Steak Sampler  \
0                        0                   1                         0   
1                        0                   1                         0   
2                        0                   1                         0   
3                        0                   1                         0   
4                        0                   1                         0   

   side_Mac and Cheese  side_Potato Salad  dessert_Blondie Brownie  
0                    1                  0                        1  
1                    1                  0                        0  
2                    0                  1                        1  
3                    0                  1                        0  
4                    0                  0                        1

In [6]:
data_d.columns

Index(['avg_rating', 'starter_Fried Chicken Tenders',
       'starter_Jumbo Shrimp Cocktail', 'starter_Sticky Chicken Tenders',
       'maindishI_BBQ Chicken', 'maindishI_Buffalo Chicken Macaroni Stacker',
       'maindishI_Pulled Pork', 'maindishII_Lobster Roll',
       'maindishII_Sausage', 'maindishII_Steak Sampler', 'side_Mac and Cheese',
       'side_Potato Salad', 'dessert_Blondie Brownie'],
      dtype='object')

In [7]:
X=data_d[['starter_Fried Chicken Tenders',
       'starter_Jumbo Shrimp Cocktail', 'starter_Sticky Chicken Tenders',
       'maindishI_BBQ Chicken', 'maindishI_Buffalo Chicken Macaroni Stacker',
       'maindishI_Pulled Pork', 'maindishII_Lobster Roll',
       'maindishII_Sausage', 'maindishII_Steak Sampler', 'side_Mac and Cheese',
       'side_Potato Salad', 'dessert_Blondie Brownie']]
y=data_d['avg_rating']

In [8]:
from sklearn.linear_model import LinearRegression
from sklearn import metrics
regressor=LinearRegression()
regressor.fit(X,y)

LinearRegression()

In [9]:
coef_df=pd.DataFrame(regressor.coef_,X.columns,columns=['Coefficient'])
coef_df

Coefficient
starter_Fried Chicken Tenders                  0.133571
starter_Jumbo Shrimp Cocktail                 -0.630588
starter_Sticky Chicken Tenders                 0.339275
maindishI_BBQ Chicken                         -0.209208
maindishI_Buffalo Chicken Macaroni Stacker     0.536821
maindishI_Pulled Pork                          0.529313
maindishII_Lobster Roll                        1.048096
maindishII_Sausage                             0.331992
maindishII_Steak Sampler                      -0.500462
side_Mac and Cheese                            0.575125
side_Potato Salad                             -0.022194
dessert_Blondie Brownie                       -0.856204

In [10]:
cost=pd.read_csv('vendor_costs.csv')
cost

Item Item Category  Cost Per Serving (Dollars)
0              Fried Chicken Tenders      Starters                        3.40
1                        Bacon Fries      Starters                        3.10
2             Sticky Chicken Tenders      Starters                        2.90
3              Jumbo Shrimp Cocktail      Starters                        4.60
4                        BBQ Brisket   Main Dish I                        6.00
5                        Pulled Pork   Main Dish I                        5.70
6                        BBQ Chicken   Main Dish I                        5.30
7   Buffalo Chicken Macaroni Stacker   Main Dish I                        6.10
8                            Sausage  Main Dish II                        4.70
9                       Lobster Roll  Main Dish II                        5.20
10                   Beef Short Ribs  Main Dish II                        6.00
11                     Steak Sampler  Main Dish II                        4.70
12                    Mac and Cheese          Side                        0.45
13                      Potato Salad          Side                        0.35
14                French Fry Platter          Side                        0.15
15                   Blondie Brownie       Dessert                        0.90
16               Apple Pie a la Mode       Dessert                        1.10

Firstly, it’s really easy to choose the starter, since “Stick Chicken Tender” can increase the most rate with the lowest cost, it’s definitely the best one. Then, for the main dish 1, we can see although the “Buffalo Chicken Macaroni Stacker” has the highest coefficient, “Pulled Pork” has a quite similar high coefficient, and with a lower cost than “Buffalo Chicken Macaroni Stacker”. So we’d love to choose “Pulled Pork” for Lobster Land. Besides, for the main dish 2, here are two popular options, “Lobster Roll” and “Sausage”. However, when we go to the cost, we think that “Lobster Roll” is not very suitable because of its high cost. Given our limited budget, we will recommend “Sausage” for Lobster Land. In addition, for the side, the coefficient of “Mac and Cheese” is much higher than others, and it’s price is reasonable, so I choose it to better satisfy visitors. Finally, for dessert, we have two choices. although “Blondie Brownie” has a lower cost, but we avoid to choose any food with a negative coefficient, so we choose “Apple Pie a la Mode” for Lobster land.

In all, the menu our group choose is “Stick Chicken Tender”, “Pulled Pork”, “Sausage”, “Mac and Cheese”, and “Apple Pie a la Mode”, the total unit cost is 14.85, and the profit is 0.15. Although the profit per visitor is not high, our group believes that if Lobster Land offers good food, it can attract more people to participate and increase the total profit. Also, I think we should not only focus on the current profit, instead, we can regard it as a great opportunity to promote Lobster Land. If visitors have a good experience this time, they are willing to come again next time and recommend it to their friends. It will definitely have long-term benefits on the development of Lobster Land.